ompi/mpi/c/status_get_error.c

```diff
@@ -0,0 +1,61 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil -*- */
+/*
+ * Copyright (c) 2025      Advanced Micro Devices, Inc. All rights reserved.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+#include "ompi_config.h"
+#include <stdio.h>
+#include <limits.h>
+
+#include "ompi/mpi/c/bindings.h"
+#include "ompi/runtime/params.h"
+#include "ompi/errhandler/errhandler.h"
+#include "ompi/memchecker.h"
+
+#if OMPI_BUILD_MPI_PROFILING
+#if OPAL_HAVE_WEAK_SYMBOLS
+#pragma weak MPI_Status_get_error = PMPI_Status_get_error
+#endif
+#define MPI_Status_get_error PMPI_Status_get_error
+#endif
+
+static const char FUNC_NAME[] = "MPI_Status_get_error";
+
+
+int MPI_Status_get_error(const MPI_Status *status, int *error)
+{
+    int rc      = MPI_SUCCESS;
+
+    MEMCHECKER(
+               if (status != MPI_STATUSES_IGNORE) {
+                   /*
+                    * Before checking the complete status, we need to reset the definedness
+                    * of the MPI_ERROR-field (single-completion calls wait/test).
+                    */
+                   opal_memchecker_base_mem_defined((void*)&status->MPI_ERROR, sizeof(int));
+                   memchecker_status(status);
+                   memchecker_datatype(datatype);
```

Comment:
what datatype ?

Reply:
@bosilca thanks for catching it.  Would it be possible to add a test to the CI that triggers this path? This is one of the items that CI should have caught in my opinion,